source/faq.txt

```diff
@@ -261,46 +261,53 @@ socket connections.
 If you experience unexpected network behavior, here
 are some things to check:
 
-#. The firewall should send a ``FIN packet`` when closing a socket,allowing the driver to detect that the socket is closed.
+#. The firewall should send a ``FIN packet`` when closing a socket,
+   allowing the driver to detect that the socket is closed.
 #. The firewall should allow ``keepAlive`` probes.
 
 How Can I Prevent a Slow Operation From Delaying Other Operations?
 ------------------------------------------------------------------
 
-You can prevent an operation from slowing down your application by
-specifying the size of the connection pool, which is the cache of
-connections that the driver maintains at any time. By increasing the
-maximum connection pool size, you can reduce application latency.
-
-To control the maximum size of a connection pool, you can set the
-``maxPoolSize`` option in the :ref:`connection options
-<node-connection-options>`. The default value of ``maxPoolSize`` is
-``100``. If the number of in-use connections to a server reaches
-``maxPoolSize``, the next request to that server will wait 
-until a connection becomes available. The following code sets
+Before you make changes to your connection configuration or application,
+you should determine which operations could be causing delays. You can
```

Comment:
Suggestion:

I think this is a pretty good approach, but seems slightly out of order because it mentions the solution ("make changes to your connection configuration or application") before explaining how to determine whether the problem is applicable to the reader.

I took a look back at the original FAQ entry referenced in [DOCSP-29272](https://jira.mongodb.org/browse/DOCSP-29272) because it was a little unclear to me whether the "other operations" in the title was referring to MongoDB operations.
 
I think, unless you add it to the title of the FAQ entry, the first paragraph needs to clarify this.

I think that the delay this is referring to can only exist when you're sharing a single MongoClient (one connection pool) and making concurrent asynchronous calls.

Here's one way I can think of to present the info:

E.g.
"
When you run multiple MongoDB operations concurrently, a slow operation can cause delays to other operations. The slow operation keeps connections to MongoDB busy which can cause other operations to wait for a free connection.

If you suspect that slow MongoDB operations are delaying other ones, you can check each operation's performance using the following methods:

- Enable the database profiler...
- ...

"